### PR TITLE
fix(firestore): More fixes to `docChanges()`

### PR DESCRIPTION
**Pull request by wSedlacek**
Extends upon the changes made by @gustavohenke
Now includes removed items when a new item is moved to an index that replaces it.

fixes: #18
Signed-off-by: William Sedlacek <wsedlacekc@gmail.com>

### CODE REVIEW

**Comment by wSedlacek** [comment]
@gustavohenke Hope you don't mind, I took it upon myself to rework the docChanges a bit and this now passes all the test.
I also added a additional test to prevent regressions to the previous behavior I observed in my last comment in #18

File: firestore/query-snapshot.ts [hunk 1]
```diff
@@ -37,54 +37,41 @@ export class MockQuerySnapshot<T = firebase.firestore.DocumentData>
...
+          type: "modified",
           oldIndex,
           newIndex,
           doc,
         });
-      } else if (oldIndex !== newIndex || !previousDocs[oldIndex].isEqual(this.docs[newIndex])) {
-        // Different index or not the same content anymore? It's a change.
+      }
+
+      if (previousDoc) checkedIDs.add(previousDoc.id);
```
Comment (gustavohenke) [inline firestore/query-snapshot.ts:63]: I noticed that this wouldn't have passed prettier lint and added it to `master`.
Can you update your branch and reformat files, please?

`npm run lint -- --write` in the root of the repo should do it.

File: firestore/query-snapshot.ts [hunk 2]
```diff
@@ -37,54 +37,41 @@ export class MockQuerySnapshot<T = firebase.firestore.DocumentData>
...
+    }
+
+    for (const [oldIndex, doc] of previousDocs.entries()) {
+      if (!checkedIDs.has(doc.id)) {
```
Comment (gustavohenke) [inline firestore/query-snapshot.ts:67]: `CollectionGroup` can also yield `QuerySnapshot`s, so I'd prefer if this check was based on `doc.path` instead.

If `foo/a` is untouched/modified and `bar/a` has been deleted, then `bar/a` won't be reported as such.

File: firestore/query-snapshot.ts (hunk 1 above)
Comment (wSedlacek) [reply firestore/query-snapshot.ts:63]: Done.

File: firestore/query-snapshot.ts (hunk 2 above)
Comment (wSedlacek) [reply firestore/query-snapshot.ts]: Wrote a test for this condition and updated to check paths rather then ID.

**Comment by wSedlacek** [comment]
No problem! Glad I was able to help! Think we could make a release to NPM?

**Comment by gustavohenke** [comment]
Just done 🚢 0.5.3 is out!

**Pull request merged**